adm_program/system/classes/phpini.php

```diff
@@ -0,0 +1,201 @@
+        $unit  = strtoupper(substr($data, -1));
+
+        switch ($unit)
+        {
+            case 'T':
+                $value *= $multi;
+            case 'G':
+                $value *= $multi;
+            case 'M':
+                $value *= $multi;
+            case 'K':
+                $value *= $multi;
+        }
+
+        return (int) $value;
+    }
+
+    /**
+     * @deprecated 3.3.0:4.0.0 This function will be removed if PHP 5.3 support gets dropped
+     * @return bool
+     */
+    public static function isSafeModeEnabled()
+    {
+        return (bool) ini_get('safe_mode');
+    }
+
+    /**
+     * @return string[]
+     */
+    public static function getBaseDirs()
+    {
+        return explode(PATH_SEPARATOR, ini_get('open_basedir'));
+    }
+
+    /**
+     * @return int
+     */
+    public static function getMemoryLimit()
+    {
+        return self::getBytesFromSize(ini_get('memory_limit'));
+    }
+
+    /**
+     * @return int
+     */
+    public static function getPostMaxSize()
+    {
+        return self::getBytesFromSize(ini_get('post_max_size'));
+    }
+
+    /**
+     * @return bool
+     */
+    public static function isFileUploadEnabled()
+    {
+        return (bool) ini_get('file_uploads');
+    }
+
+    /**
+     * @return string
+     */
+    public static function getFileUploadTmpDir()
+    {
+        return ini_get('upload_tmp_dir');
+    }
+
+    /**
+     * @return int
+     */
+    public static function getFileUploadMaxFileSize()
+    {
+        return self::getBytesFromSize(ini_get('upload_max_filesize'));
+    }
+
+    /**
+     * @return int
+     */
+    public static function getFileUploadMaxFileCount()
+    {
+        return (int) ini_get('max_file_uploads');
+    }
+
+    /**
+     * @return int
+     */
+    public static function getUploadMaxSize()
+    {
+        return min(self::getMemoryLimit(), self::getPostMaxSize(), self::getFileUploadMaxFileSize());
+    }
+
+    /**
+     * @return bool
+     */
+    public static function checkSizeLimits()
+    {
+        return (self::getMemoryLimit() === -1 || self::getMemoryLimit() >= self::getPostMaxSize())
+            && (self::getPostMaxSize() === -1 || self::getPostMaxSize() >= self::getFileUploadMaxFileSize());
+    }
+
+    /**
+     * @param string $dirPath
+     * @return bool
+     */
+    private static function isInBaseDirs($dirPath)
+    {
+        $baseDirs = self::getBaseDirs();
+
+        if ($baseDirs[0] === '')
+        {
+            return true;
+        }
+
+        $isInBaseDirs = false;
+        foreach ($baseDirs as $baseDir)
+        {
+            if (strpos($dirPath, $baseDir) === 0)
+            {
+                $isInBaseDirs = true;
+            }
+        }
+
+        return $isInBaseDirs;
+    }
+
+    /**
+     * @param string[] $dirPaths
+     * @return bool|string
+     * @throws Exception
+     */
+    public static function setBaseDirs(array $dirPaths = array())
```

Comment:
This method is not called yet !?!
Please add an description of the method.

Reply:
I already have plans for this function in the future.